### DISCUSSION TITLE
while using new model like ch_PP-OCRv4_det_infer.onnx ，error

### DISCUSSION THREAD

**Discussion by Kyrie666** [RapidAI/TableStructureRec#32] [Q&A]
```python
import os

from lineless_table_rec import LinelessTableRecognition
from lineless_table_rec.utils_table_recover import format_html, plot_rec_box_with_logic_info, plot_rec_box
from table_cls import TableCls
from wired_table_rec import WiredTableRecognition

lineless_engine = LinelessTableRecognition()
wired_engine = WiredTableRecognition()
table_cls = TableCls()
img_path = f'images/img14.jpg'

cls,elasp = table_cls(img_path)
if cls == 'wired':
    table_engine = wired_engine
else:
    table_engine = lineless_engine
model_path = os.getcwd() + "/model/ch_PP-OCRv4_det
html, elasp, polygons, logic_points, ocr_res = table_engine(img_path,ocr_result=model_path)
print(f"elasp: {elasp}")

# output_dir = f'outputs'
# complete_html = format_html(html)
# os.makedirs(os.path.dirname(f"{output_dir}/table.html"), exist_ok=True)
# with open(f"{output_dir}/table.html", "w", encoding="utf-8") as file:
#     file.write(complete_html)
# # 可视化表格识别框 + 逻辑行列信息
# plot_rec_box_with_logic_info(
#     img_path, f"{output_dir}/table_rec_box.jpg", logic_points, polygons
# )
# # 可视化 ocr 识别框
# plot_rec_box(img_path, f"{output_dir}/ocr_box.jpg", ocr_res)
```

**Comment by Kyrie666** [comment]
the model file is correct ,ignoring writing err

**Reply by Joker1212** [reply]
if you want to use higher pricision ocr model ,this is the right way ,ocr result means the output of ocr model 
ocr_engine =RapidOCR(det_model_dir="det_server_infer.onnx",rec_model_dir="rec_server_infer.onnx")
ocr_res, _ = ocr_engine(img_path)
html, elasp, polygons, logic_points, ocr_res = table_engine(img_path, ocr_result=ocr_res)

**Comment by Kyrie666** [comment]
Sorry, I didn't analyze the source code. Now I have used the correct calling method. Thank you.



---Original---
From: ***@***.***&gt;
Date: Thu, Sep 26, 2024 16:11 PM
To: ***@***.***&gt;;
Cc: "Kyrie ***@***.******@***.***&gt;;
Subject: Re: [RapidAI/TableStructureRec] while using new model like ch_PP-OCRv4_det_infer.onnx ，error (Discussion #32)




 
if you want to use higher pricision ocr model ,this is the right way ,ocr result means the output of ocr model
 ocr_engine =RapidOCR(det_model_dir="det_server_infer.onnx",rec_model_dir="rec_server_infer.onnx")
 ocr_res, _ = ocr_engine(img_path)
 html, elasp, polygons, logic_points, ocr_res = table_engine(img_path, ocr_result=ocr_res)
